ports/gmp/portfile.cmake

```
@@ -2,116 +2,78 @@ if(EXISTS "${CURRENT_INSTALLED_DIR}/include/gmp.h" OR "${CURRENT_INSTALLED_DIR}/
+    set(ENV{CC_FOR_BUILD} "touch a.out | touch conftest${VCPKG_HOST_EXECUTABLE_SUFFIX} | true")
+    set(ENV{CPP_FOR_BUILD} "touch a.out | touch conftest${VCPKG_HOST_EXECUTABLE_SUFFIX} | true")
+endif()
 
-    vcpkg_install_msbuild(
-        USE_VCPKG_INTEGRATION
-        SOURCE_PATH ${SOURCE_PATH}
-        PROJECT_SUBPATH SMP/libgmp.sln
-        PLATFORM ${TRIPLET_SYSTEM_ARCH}
-        LICENSE_SUBPATH COPYING.LESSERv3
-        TARGET Rebuild
-        RELEASE_CONFIGURATION ${CONFIGURATION_RELEASE}
-        DEBUG_CONFIGURATION ${CONFIGURATION_DEBUG}
-        SKIP_CLEAN
-        OPTIONS "/p:YasmPath=${YASM}"
-    )
-    get_filename_component(SOURCE_PATH_SUFFIX "${SOURCE_PATH}" NAME)
-    file(RENAME "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel/${SOURCE_PATH_SUFFIX}/msvc/include" "${CURRENT_PACKAGES_DIR}/include")
-    set(PACKAGE_VERSION 6.2.1)
-    set(PACKAGE_NAME gmp)
-    set(prefix "${CURRENT_INSTALLED_DIR}")
-    set(exec_prefix "\${prefix}")
-    set(libdir "\${prefix}/lib")
-    set(includedir "\${prefix}/include")
-    set(LIBS -lgmp)
-    configure_file("${SOURCE_PATH}/gmp.pc.in" "${CURRENT_PACKAGES_DIR}/lib/pkgconfig/gmp.pc" @ONLY)
-    configure_file("${SOURCE_PATH}/gmpxx.pc.in" "${CURRENT_PACKAGES_DIR}/lib/pkgconfig/gmpxx.pc" @ONLY)
-    set(prefix "${CURRENT_INSTALLED_DIR}/debug")
-    set(exec_prefix "\${prefix}")
-    set(libdir "\${prefix}/lib")
-    set(includedir "\${prefix}/../include")
-    set(LIBS -lgmpd)
-    configure_file("${SOURCE_PATH}/gmp.pc.in" "${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig/gmp.pc" @ONLY)
-    configure_file("${SOURCE_PATH}/gmpxx.pc.in" "${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig/gmpxx.pc" @ONLY)
-    vcpkg_fixup_pkgconfig()
-    if(VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
-        vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/gmp.h"
-                            "#if defined(DLL_EXPORT) && defined(NO_ASM)"
-                            "#if 1")
-    endif()
-else()
-    vcpkg_download_distfile(
-        ARCHIVE
-        URLS https://gmplib.org/download/gmp/gmp-6.2.1.tar.xz
-        FILENAME gmp-6.2.1.tar.xz
-        SHA512 c99be0950a1d05a0297d65641dd35b75b74466f7bf03c9e8a99895a3b2f9a0856cd17887738fa51cf7499781b65c049769271cbcb77d057d2e9f1ec52e07dd84
-    )
+vcpkg_configure_make(
+    SOURCE_PATH "${SOURCE_PATH}"
+    AUTOCONFIG
+    OPTIONS
+        ${OPTIONS}
+        --enable-cxx
+)
 
-    vcpkg_extract_source_archive_ex(
-        OUT_SOURCE_PATH SOURCE_PATH
-        ARCHIVE ${ARCHIVE}
-        REF gmp-6.2.1
-        PATCHES
-            tools.patch
-    )
+set(tool_names bases fac fib jacobitab psqr trialdivtab)
+list(TRANSFORM tool_names PREPEND "gen-")
+list(TRANSFORM tool_names APPEND "${VCPKG_HOST_EXECUTABLE_SUFFIX}")
 
-    vcpkg_configure_make(
-        SOURCE_PATH ${SOURCE_PATH}
-        AUTOCONFIG
-        OPTIONS
-            --enable-cxx
-    )
+if(VCPKG_CROSSCOMPILING)
+    list(TRANSFORM tool_names PREPEND "${CURRENT_HOST_INSTALLED_DIR}/manual-tools/${PORT}/")

```

```suggestion
    list(TRANSFORM tool_names PREPEND "${CURRENT_HOST_INSTALLED_DIR}/tools/${PORT}/")
```
I don't think this is special enough to need a new top-level directory "manual-tools"

---

Manual-tools already exists in e.g. the vcpkg-tool-python2 port and it is intended that those tools dont get automatically added to CMAKE_PROGRAM_PATH.